awscli/autocomplete/parser.py

```diff
@@ -150,6 +150,15 @@ def parse(self, command_line, location=None):
 
     def _consume_value(self, remaining_parts, option_name,
                        lineage, current_command):
+        # We have a special case where a user is trying to complete
+        # a value for an option, which is the last fragment of the command,
+        # e.g. 'aws ec2 describe-instances --instance-ids '
+        # Note the space at the end.  In this case we don't have a value
+        # to consume so we special case this and short circuit.
+        if remaining_parts == ['']:
```

Comment:
It may be worth moving the empty string into a constant to help with readability and to help indicate its purpose in the parsing logic. Not sure about what name to give it though. Looking over the tests it would especially help in reading the difference between empty string and `None` for the `last_fragment`.